p2p/gossip/protocol.go

```diff
@@ -91,21 +91,31 @@ func (prot *Protocol) markMessageAsOld(h types.Hash12) bool {
 }
 
 func (prot *Protocol) propagateMessage(payload []byte, h types.Hash12, nextProt string, exclude p2pcrypto.PublicKey) {
+	//TODO soon : don't wait for mesaage to send and if we finished sending last message one of the peers send the next message to him.
+	// limit the number of simultaneous sends. *consider other messages (mainly sync)
 	prot.peersMutex.RLock()
-peerLoop:
+	peers := make([]p2pcrypto.PublicKey, 0, len(prot.peers))
```

Comment:
consider refactoring to getPeers func

Reply:
this pattern and usage of peers also takes place in `sync` so I'd rather do it in a dedicated pr